README.md

```diff
@@ -67,15 +69,15 @@ trick. This might be beneficial for aggregate roots that have a lot of events
 (think thousands, or more).
 
 ## Event dispatcher
-The aggregate root has an event dispatcher that by default uses compiled
+Each aggregate has an event dispatcher that by default uses compiled
 expressions to execute matching non-public `When(@event)` methods. This is
 extremely fast, but if desired, by implementing a own `EventDispatcher`, the
 behavior can be changed.
 
 ## Example 1
-A (simplified) real life example of a financial entry, using `AggregateRoot<TAggregate, TId>`:
+A (simplified) real life example of a financial entry, using `Aggregate<TAggregate, TId>`:
 ``` C#
-public sealed class FinancialEntry : AggregateRoot<FinancialEntry, Guid>
+public sealed class FinancialEntry : Aggregate<FinancialEntry, Guid>
 {
     // Note that FinancialEntryValidator is passed in for validation purposes.
     // See the implementation of this class below.
```

Comment:
I can't put a comment there, but there is a typo in line 158/160 "`Result<TAggregate>` is up to the developer"